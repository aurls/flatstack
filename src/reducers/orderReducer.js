import actionTypes from '../constants/actionTypes';
import stages from '../constants/stages';

export const initialState = {
  stage: stages.SHIPPING,
  shipping: {
    name: '',
    phone: '',
    street: '',
    streetMore: '',
    city: '',
    country: '',
    zip: '',
    errors: {}
  },
  billing: {
    name: '',
    email: '',
    street: '',
    streetMore: '',
    city: '',
    country: '',
    zip: '',
    errors: {}
  },
  payment: {
    cardholder: '',
    cardNumber: '',
    expireDate: '',
    securityCode: '',
    errors: {}
  }
};

const orderReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case actionTypes.SET_ORDER_STAGE:
      return setOrderStage(state, payload);

    case actionTypes.UPDATE_SHIPPING:
      return updateShipping(state, payload);

    case actionTypes.SET_SHIPPING_LOCATION:
      return setShippingLocation(state, payload);

    case actionTypes.UPDATE_BILLING:
      return updateBilling(state, payload);

    case actionTypes.SET_BILLING_LOCATION:
      return setBillingLocation(state, payload);

    case actionTypes.UPDATE_PAYMENT:
      return updatePayment(state, payload);

    default:
      return state;
  }
};

function setOrderStage (state, nextStage) {
  return {
    ...state,
    stage: nextStage
  };
}

function updateShipping (state, newData) {
  return {
    ...state,
    shipping: {
      ...state.shipping,
      ...newData
    }
  }
}

function setShippingLocation (state, location) {
  return {
    ...state,
    shipping: {
      ...state.shipping,
      city: location.city,
      country: location.country,
      zip: location.zip
    }
  }
}

function updateBilling (state, newData) {
  return {
    ...state,
    billing: {
      ...state.billing,
      ...newData
    }
  }
}

function setBillingLocation (state, location) {
  return {
    ...state,
    billing: {
      ...state.billing,
      city: location.city,
      country: location.country,
      zip: location.zip
    }
  }
}

function updatePayment (state, newData) {
  return {
    ...state,
    payment: newData
  }
}

export default orderReducer;
