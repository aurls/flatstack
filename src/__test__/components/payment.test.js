import React from 'react';
import Payment from '../../components/payment';
import { payment } from '../reducers/orderReducer.test';

describe('Payment component', () => {
  let component;
  const setOrderStage = jest.fn();
  const updatePayment = jest.fn();

  beforeEach(() => {
    component = shallow(
      <Payment
        payment={payment}
        setOrderStage={setOrderStage}
        updatePayment={updatePayment} />
    );
  });

  it('Default state', () => {
    expect(component).toMatchSnapshot();
  });
});
