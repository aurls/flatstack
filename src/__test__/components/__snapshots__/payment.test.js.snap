// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment component Default state 1`] = `
<form
  className="payment"
  name="payment"
>
  <h2
    className="order-info__title"
  >
    Payment
  </h2>
  <p
    className="payment__security-disclaimer"
  >
    <img
      alt=""
      className="payment__security-disclaimer-icon"
      src={null}
    />
    This is a secure 128-bit SSL encrypted payment
  </p>
  <h3
    className="order-info__label"
  >
    Cardholder Name
  </h3>
  <OrderInput
    areErrorsVisible={false}
    autofocus={true}
    errors={Object {}}
    handleInput={[Function]}
    inputMode="text"
    maxLength={null}
    name="cardholder"
    placeholder="Name as it appears on your card"
    type="text"
    value="John Dow"
  />
  <h3
    className="order-info__label"
  >
    Card Number
  </h3>
  <OrderInput
    areErrorsVisible={false}
    autofocus={false}
    errors={Object {}}
    handleInput={[Function]}
    inputMode="numeric"
    maxLength={null}
    name="cardNumber"
    placeholder="XXXX XXXX XXXX XXXX"
    type="text"
    value="4200 1234 5678"
  >
    <img
      alt=""
      className="payment__card-logo"
      src={null}
    />
  </OrderInput>
  <div
    className="order-info__row"
  >
    <div>
      <h3
        className="order-info__label"
      >
        Expire Date
      </h3>
      <OrderInput
        areErrorsVisible={false}
        autofocus={false}
        errors={Object {}}
        handleInput={[Function]}
        inputMode="numeric"
        maxLength={null}
        name="expireDate"
        placeholder="MM / YY"
        type="text"
        value="12 / 21"
      />
    </div>
    <div>
      <h3
        className="order-info__label"
      >
        Security Code
      </h3>
      <OrderInput
        areErrorsVisible={false}
        autofocus={false}
        errors={Object {}}
        handleInput={[Function]}
        inputMode="numeric"
        maxLength={3}
        name="securityCode"
        placeholder=""
        type="password"
        value="432"
      />
    </div>
  </div>
  <div
    className="order-info__continue"
    onClick={[Function]}
  >
    Pay Securely
  </div>
</form>
`;
